src/pickup.cpp

```diff
@@ -251,6 +251,116 @@ static bool select_autopickup_items( std::vector<item> &here, std::vector<bool>
     return bFoundSomething;
 }
 
+
+// parameter definition - append_fail_msg - any other message you want to append to the error message displayed
+//                      - silent_chk - if true, will not display any error message
+//                      - allow_swap - if true, when exceed volume, will always return true (unless current
+//                                     wielded weapon is no_unwield flag set, else return false
+//                      - chk_keep_hands_free - if true, will check for keep_hands_free. Note that this will not have any effect if not
+//                                               allow_swap
+//                      - chk_armor - if true, when item is armor and exceed volume, will return true as player might want to wear it
+bool Pickup::can_pick_one_up( player &p, item &newit, std::string append_fail_msg, bool silent_chk,
+                              bool allow_swap, bool chk_keep_hands_free, bool chk_armor )
+{
+    // if item is already in inventory or is weapon
+    if ( p.inv.has_item(&newit) || (p.has_weapon() && &p.weapon == &newit) ) {
+        return true;
+    }
+
+    // if item is already worn
+    for( auto &w : p.worn ) {
+        if ( &w == &newit ) {
+            return true;
+        }
+    }
+
+    if ( !p.can_pickup(false) ) {
+        if ( p.is_player() && !silent_chk ) {
+            add_msg( m_info, _(("You cannot pick up items with your %s! " + append_fail_msg).c_str()), p.weapon.tname().c_str() );
```

Comment:
> _(("There's no room in your inventory for the %s. " + append_fail_msg)

This prevents the extraction of the string literal for translation. (String literals in `_()` are automatically extracted for translation by the script "lang/po/cataclysm-dda.pot" and during runtime the `_` function translates the string.)

Additionally, translators are known to forget to add trailing/leading spaces. I suggest you remove the `append_fail_msg` parameter and simply add the additional message in game.cpp when `can_pick_one_up` returns `false`.